common/src/main/kotlin/fr/agrouagrou/common/turn/WitchActions.kt

```diff
@@ -0,0 +1,64 @@
+package fr.agrouagrou.common.turn
+
+import fr.agrouagrou.common.GameState
+import fr.agrouagrou.common.player.Player
+import fr.agrouagrou.common.player.PlayerManager
+import fr.agrouagrou.common.player.Role
+import kotlinx.coroutines.flow.MutableStateFlow
+import java.util.*
+
+class WitchActions(
+    private val gameState: MutableStateFlow<GameState>,
+    private val playerManager: PlayerManager,
+) : RoleActions<Role.Witch>(Role.Witch::class, GameState.NIGHT_WITCH) {
+    private var hasUsedPotion = false
+
+    fun killPlayer(
+        playerId: String,
+        targetId: String,
+    ) {
+        val (player, isTargetAlive) = prepareUsingPotion(playerId, targetId)
+
+        if (!isTargetAlive) {
+            throw IllegalArgumentException("Target is already dead")
+        }
+
+        (player.role as Role.Witch).useDeathPotion()
+        playerManager.setPlayerAlive(UUID.fromString(targetId), false)
+        hasUsedPotion = true
+    }
+
+    fun revivePlayer(
+        playerId: String,
+        targetId: String,
+    ) {
+        val (player, isTargetAlive) = prepareUsingPotion(playerId, targetId)
+
+        if (isTargetAlive) {
+            throw IllegalArgumentException("Target is already alive")
+        }
+
+        (player.role as Role.Witch).useLifePotion()
+        playerManager.setPlayerAlive(UUID.fromString(targetId), true)
+        hasUsedPotion = true
+    }
```

Comment:
how to check if the player was killed on the current turn? maybe three states (alive, vegetable, dead) more than a boolean?

because the Witch can revive only a player killed on the current turn

Reply:
needs to change the field `alive` to an enum?

Reply:
Done! I implemented an enum as you suggested.